src/Integration/Notifications/SonarQubeNotifications.cs

```diff
@@ -0,0 +1,145 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Threading.Tasks;
+using SonarQube.Client.Models;
+using SonarQube.Client.Services;
+using CancellationTokenSource = System.Threading.CancellationTokenSource;
+
+namespace SonarLint.VisualStudio.Integration.Notifications
+{
+    [Export(typeof(ISonarQubeNotifications))]
+    internal class SonarQubeNotifications : ISonarQubeNotifications
+    {
+        private readonly ITimer timer;
+        private readonly ISonarQubeService sonarQubeService;
+        private readonly CancellationTokenSource cancellation = new CancellationTokenSource();
+
+        private DateTimeOffset lastCheckDate;
+        private string projectKey;
+
+        public INotificationIndicatorViewModel Model { get; private set; }
+
+        public NotificationData GetNotificationData() =>
+            new NotificationData
+                {
+                    IsEnabled = Model.AreNotificationsEnabled,
+                    LastNotificationDate = lastCheckDate
+                };
+
+        [ImportingConstructor]
+        [ExcludeFromCodeCoverage] // Do not unit test MEF constructor
+        internal SonarQubeNotifications(IHost host)
+            : this(host.SonarQubeService, new NotificationIndicatorViewModel(),
+                  new TimerWrapper { Interval = 60000 })
+        {
+        }
+
+        internal SonarQubeNotifications(ISonarQubeService sonarQubeService,
+            INotificationIndicatorViewModel model, ITimer timer)
```

Comment:
I still don't like taking `ITimer` as argument instead of `ITimerFactory` but....